pkg/git/clone.go

```diff
@@ -26,14 +27,13 @@ func (c *Clone) Download(config *api.Config) error {
 		glog.V(2).Infof("Cloning into %s", targetSourceDir)
 		if err := c.Clone(config.Source, targetSourceDir); err != nil {
 			glog.V(1).Infof("Git clone failed: %+v", err)
-			return err
+			return nil, err
 		}
 
 		if config.Ref != "" {
 			glog.V(1).Infof("Checking out ref %s", config.Ref)
-
 			if err := c.Checkout(targetSourceDir, config.Ref); err != nil {
-				return err
+				return c.GetInfo(targetSourceDir), err
```

Comment:
no, we return only if the checkout ref fails ;-) ufffff (and I just removed the GetInfo() if that happens)